### PR TITLE
Tenant-scoped Case Handover co-access and expiry

### LINKED ISSUE #200: [CAR-CHO-01] Access outcomes: co-access vs takeover + reclaim + co-access auto-expiry
Affected repos: ORISO-UserService

## Why

The current `requestAccess` path transfers ownership for every Case Handover reason. ADR-002 requires a finite read-only co-access outcome for Advice Needed and a non-expiring takeover outcome for absence reasons.

## What to build

- Implement Advice Needed as read-only co-access using the existing Supervision/session-supervisor primitives; ownership and the original counsellor's visibility remain unchanged.
- Expire co-access with an injected clock, revoke it automatically, and record structured grant/expiry/extension audit events.
- Validate durations as non-null, minimum 15 minutes, 15-minute increments, default 180 minutes, and without a product maximum.
- Keep takeover reasons non-expiring (`null` duration), transfer ownership without evicting the previous counsellor from the room, and support reclaim except for permanent reasons.
- Restrict all access outcomes to eligible counsellors in the same agency and integrate the team-access consent floor from #313.

## Acceptance

- [ ] Advice Needed grants read-only co-access, preserves ownership, and expires exactly under an injected clock.
- [ ] 15, 180, and large 15-minute-multiple durations work; null/below-min/non-multiples fail for Advice Needed.
- [ ] Takeover has no TTL, preserves room membership, and supports the documented reclaim rules.
- [ ] Grant, expiry, extension, takeover, and reclaim create structured audit entries.
- [ ] Cross-agency requests are denied without exposing session data.

### Reviewer test plan

- [ ] Request Advice Needed for 15 minutes → the second counsellor receives read-only access and ownership stays unchanged.
- [ ] Advance the test clock to the exact expiry → access is revoked once and one expiry audit event exists.
- [ ] Request 180 minutes and a large valid duration → both persist; null, 14, and 16 minutes are rejected.
- [ ] Complete an absence takeover → ownership changes, the previous counsellor remains a silent room member, and no expiry is scheduled.
- [ ] Attempt access from a counsellor in another agency → receive 403 with no session details.

### LINKED ISSUE #201: [CAR-CHO-01] Move handover policy config to TenantService + expand policy model
Affected repos: ORISO-UserService

## Why

The existing `case_handover_reason_policy` is a global enforcement table. TenantService will become the canonical owner; UserService must keep a tenant-scoped effective cache and enforce it without recreating a second configuration source.

## What to build

- Replace mutable global policy ownership with a tenant-scoped effective cache populated from TenantService's typed policy contract.
- Cache/enforce approval-role sets, multilingual templates, and duration policies; keep the legacy table/shape dual-readable for the transition release.
- Make cache refresh/startup/retry behavior explicit and safe: last-known-good policy remains enforceable, stale state is observable, and tenant data never crosses boundaries.
- Enforce inherited read-only policy values in every handover gate and expose only resolved policy data to consumers.
- Update OpenAPI clients and add cache, compatibility, failure/restart, truth-table, and tenant-isolation tests.

## Acceptance

- [ ] TenantService is the only configuration owner; UserService stores only effective tenant-scoped cache data.
- [ ] Approval roles, templates, and duration are enforced from the resolved typed policy.
- [ ] Legacy rows remain readable without accepting legacy writes as the new contract.
- [ ] Cache outage/restart behavior is tested and cannot leak or reset another tenant's policy.

### Reviewer test plan

- [ ] Publish two different tenant policy snapshots → each tenant's handover gate uses only its own snapshot.
- [ ] Restart UserService with a valid last-known-good cache → enforcement remains deterministic and stale status is observable.
- [ ] Simulate a refresh failure → no permissive default replaces an enforced cached value.
- [ ] Read migrated legacy reason policies → effective values match the documented conversion and remain read-only from UserService.
- [ ] Attempt to request another tenant's resolved policy → receive 403 without policy details.

### PR DESCRIPTION
**Pull request by Storypapst**
## Summary

- consumes TenantService-owned handover policies through a tenant-keyed last-known-good cache
- implements Advice Needed as time-limited read-only co-access with 15-minute steps, 3-hour default, no product maximum, exact expiry, and audit state
- retains takeover semantics without TTL and preserves the original reason codes
- keeps explicit client consent inside the Case Handover flow
- removes the accidentally introduced standalone Team Access endpoint, DTO projection, status, and coupling to the pre-existing Supervision opt-out
- preserves the existing `/supervision/opt-out` endpoint and persisted `supervisionOptedOut` behavior unchanged

## Verification

- focused Case Handover, cache, supervision facade/controller, and mapper tests passed
- full `./mvnw -q test` on JDK 21 passed for the configured unit-test suite
- `python3 -m pytest -q tests/contracts/test_openapi_contract_gate.py` — 9 passed
- contract tests explicitly prevent the removed `/team-access` endpoint and `teamAccessAllowed` field from returning

Closes #200
Closes #201
Related to #205
Related to OpenResilienceInitiative/ORISO-Admin#771
Depends on OpenResilienceInitiative/ORISO-TenantService#193

## Summary by CodeRabbit

* **New Features**
  * Added temporary co-access and takeover handling for case handovers.
  * Added configurable consent modes, access durations, expiration details, and localized policy information.
  * Co-access now expires automatically while preserving existing counsellor session ownership.
  * Added resilient tenant policy retrieval using cached last-known-good configurations.

* **Bug Fixes**
  * Improved handling of expired access, policy-service failures, and invalid policy data.
  * Added consent details to handover notifications and status information.

* **Tests**
  * Expanded coverage for consent, expiration, caching, localization, scheduling, and access reconciliation.

### CODE REVIEW

**Comment by Storypapst** [comment]
Review remediation pushed in `b37f768c`:

- co-access revoke/expiry now reconciles Matrix membership before persisting the audit state
- failed reconciliation stays retryable; takeover remains untouched
- expiry is serialized across replicas under technical tenant context with pessimistic row locks
- pending approvals retain their captured duration even after policy changes
- policy-cache refreshes are tenant-serialized and fail closed
- controller inversion, tenant mismatch and duration/plural edge cases are covered

Verification: 3,979 Maven unit tests + 73 Python CI contract tests, 0 failures/errors.

File: src/main/java/de/caritas/cob/userservice/api/facade/SessionSupervisorFacade.java [hunk 1]
```diff
@@ -439,8 +441,9 @@ public void setSupervisionOptedOut(Long sessionId, boolean optedOut) {
 
     if (optedOut) {
       deactivateAllActiveSupervisors(session);
+      caseHandoverService.revokeActiveCoAccessForSession(sessionId);
     }
-    log.info("Session {} supervision opt-out set to {}", sessionId, optedOut);
+    log.info("Session {} team access allowed set to {}", sessionId, !optedOut);
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/main/java/de/caritas/cob/userservice/api/facade/SessionSupervisorFacade.java]: Verified against the current head and addressed by the existing hardening in this PR: `SessionSupervisorFacade.setSupervisionOptedOut` delegates to `revokeActiveCoAccessForSession`; both opt-out revocation and expiry call Matrix `removeUserFromRoom` before persisting `REVOKED`/`EXPIRED`, and failures keep the grant retryable instead of reporting it revoked. `CaseHandoverServiceTest` asserts Matrix removal for both lifecycles. Validation: 36/36 targeted handover tests, 9/9 contracts, and all PR CI checks green.

**Comment by Storypapst** [comment]
Final contract and security evidence on `34d7c910`:
- all PR CI checks green, including OpenAPI provider/consumer, MariaDB, Redis, integration and Docker gates
- 3,979 Maven tests and 73 contract tests passed in the full local gate
- 36/36 targeted case-handover tests and 9/9 contract checks passed after the final changes
- the remaining Matrix-revocation review thread was verified against current code, answered with the lifecycle evidence, and resolved

The implementation reuses the existing supervision opt-out storage and removes Matrix access before recording co-access as revoked/expired.

**Comment by Storypapst** [comment]
Contract dependency verified on 2026-08-17:

- This PR's TenantService consumer contract is fully compatible with TenantService PR #193 at `67341714dd05855fea4dfe87fbca2dd4a90b9391` using the same pinned `oasdiff v1.17.0` script as CI.
- The current failing consumer/provider check compares against the older pre-dev provider contract, where `clientConsentRequired` is still canonical.
- Do not waive or ignore this gate. Merge TenantService PR #193 first, then rerun this PR's checks.

**Comment by Storypapst** [comment]
@coderabbitai review

File: src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverPolicyCacheService.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  /**
+   * Refreshes one tenant. A failed upstream call never replaces the persisted last-known-good value
+   * with a permissive default; it marks that snapshot stale and keeps enforcing it.
+   */
+  @Transactional
+  public CaseHandoverPolicies refresh(Long tenantId) {
+    var existing = repository.findById(tenantId);
+    if (!scheduledTaskClaimService.tryClaim(
+        "case-handover-policy-refresh-" + tenantId, Duration.ofMinutes(1))) {
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverPolicyCacheService.java]: Done - the lease is now `@Value("${case.handover.policy-refresh-claim-duration:PT1M}")`, mirroring the co-access sweep knob, with a test pinning the configured duration onto the claim.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverPolicyCacheService.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      return existing
+          .map(this::deserialize)
+          .orElseThrow(
+              () -> new IllegalStateException("Case Handover policy refresh already in progress"));
+    }
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverPolicyCacheService.java:57]: Done - a lease conflict without a usable snapshot now throws `ServiceUnavailableException`, mapped to HTTP 503 in `ApiResponseEntityExceptionHandler`, so callers retry instead of seeing a 500.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/main/resources/replica-safety-components.json [hunk 4]
```diff
@@ -331,6 +331,28 @@
...
+      "id": "case-handover-policy-cache-refresh",
+      "source": "src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverPolicyCacheService.java",
+      "kind": "scheduler",
+      "owner": "agency-tenant",
+      "risk": "correctness",
+      "components": ["refreshKnownTenants"],
+      "decision": "Each tenant refresh is serialized through ScheduledTaskClaimService and keeps enforcing the persisted last-known-good snapshot.",
+      "signals": ["userservice.scheduler.executions", "userservice.scheduler.duration"],
+      "status": "bounded"
+    },
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/main/resources/replica-safety-components.json:474]: Fixed at the root instead of narrowing the wording: `getEffective` and `refreshKnownTenants` now open the transaction at the Spring proxy boundary, so the read and save inside `refresh` no longer commit separately on the production paths. Serialization itself continues to come from `ScheduledTaskClaimService`, as the inventory states.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/test/java/de/caritas/cob/userservice/api/service/CaseHandoverPolicyCacheServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  private CaseHandoverPolicyCacheService service;
+
+  @BeforeEach
+  void setUp() {
+    service =
+        new CaseHandoverPolicyCacheService(
+            repository, tenantServiceFactory, scheduledTaskClaimService, clock);
+    lenient().when(tenantServiceFactory.createControllerApi()).thenReturn(tenantControllerApi);
+    when(scheduledTaskClaimService.tryClaim(anyString(), any())).thenReturn(true);
+  }
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/test/java/de/caritas/cob/userservice/api/service/CaseHandoverPolicyCacheServiceTest.java:54]: Done - added `getEffective_refreshesTheTenantWhenNoSnapshotIsCached` driving the real consumer entry point, plus an annotation test asserting `@Transactional` sits on `getEffective`/`refreshKnownTenants`, which is where the proxy boundary actually is (see the sibling fix).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/test/java/de/caritas/cob/userservice/api/service/CaseHandoverServiceTest.java [hunk 6]
```diff
@@ -264,6 +321,144 @@ void requestAccess_keepsContentLocked_WhenPolicyRequiresClientConsent() {
...
+            CaseHandoverRequest.Status.GRANTED,
+            CaseHandoverRequest.AccessType.CO_ACCESS,
+            LocalDateTime.of(2026, 8, 16, 10, 0)))
+        .thenReturn(List.of(failingRequest));
+
+    assertEquals(0, caseHandoverService.expireCoAccess());
+
+    assertEquals(CaseHandoverRequest.Status.GRANTED, failingRequest.getStatus());
+    verify(caseHandoverRequestRepository).saveAll(List.of());
+  }
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/test/java/de/caritas/cob/userservice/api/service/CaseHandoverServiceTest.java:832]: Done - the sweep test now carries a second, healthy request behind the failing one and asserts it still expires: `expireCoAccess` returns 1 and `saveAll` contains only the healthy request. Deleting the `return` in the forEach now fails the test.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverService.java [hunk 7]
```diff
@@ -345,22 +433,31 @@ public CaseHandoverStatus requestAccess(Long sessionId, String reasonCode, Strin
...
-      session.setUpdateDate(now);
-      sessionRepository.save(session);
+      if (request.getAccessType() == AccessType.TAKEOVER) {
+        session.setConsultant(requester);
+        session.setUpdateDate(now);
+        sessionRepository.save(session);
+      }
       notifyGranted(saved);
+      if (status == Status.GRANTED_PENDING_CLIENT_OPTOUT) {
+        notifyPendingConsent(saved);
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverService.java:536]: Addressed with rollback compensation: the join records whether the requester was already a room member and registers a transaction synchronization that removes the membership again if the grant rolls back - never touching pre-existing ADR-002 department memberships. A full provisioning state machine was deliberately avoided; three new tests cover rollback-removes, commit-keeps, and preexisting-member-untouched.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/main/resources/db/changelog/changeset/0088_case_handover_consent_mode/0088_changeSet.xml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      SET client_consent_mode = CASE WHEN client_consent_required = 1 THEN 'OPT_IN' ELSE 'NONE' END;
+      UPDATE case_handover_request
+      SET client_consent_mode = CASE WHEN client_consent_required = 1 THEN 'OPT_IN' ELSE 'NONE' END;
+      UPDATE case_handover_reason_policy
+      SET enabled = 0, access_allowed = 0
+      WHERE code = 'OTHER_EMERGENCY';
+    </sql>
+    <rollback>
+      <dropColumn tableName="case_handover_request" columnName="client_consent_mode" />
+      <dropColumn tableName="case_handover_reason_policy" columnName="client_consent_mode" />
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/main/resources/db/changelog/changeset/0088_case_handover_consent_mode/0088_changeSet.xml:46]: Done - the UPDATE now only retires `OTHER_EMERGENCY` rows still carrying the untouched 0057/0059 seed values (admin-edited rows are never overwritten), and the rollback restores exactly that seeded default before dropping the consent columns.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/test/java/de/caritas/cob/userservice/api/service/CaseHandoverServiceTest.java [hunk 9]
```diff
@@ -111,12 +125,160 @@ void setUp() {
...
+            .build();
+    when(caseHandoverRequestRepository.findByIdAndSessionId(101L, 123L))
+        .thenReturn(Optional.of(request));
+
+    var status = caseHandoverService.resolveClientConsent(123L, 101L, false);
+
+    assertEquals("CLIENT_CONSENT_DECLINED", status.getStatus());
+    assertFalse(status.isCanViewContent());
+    assertEquals("CLIENT_CONSENT_DECLINED", status.getAuditOutcome());
+  }
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/test/java/de/caritas/cob/userservice/api/service/CaseHandoverServiceTest.java:261]: Done - the test now configures the Matrix room and identities, stubs member lookup, operator token and removal, and verifies `removeUserFromRoom`; removing the revocation call fails the test.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverService.java [hunk 10]
```diff
@@ -798,7 +1188,18 @@ private void notifyGranted(CaseHandoverRequest request) {
...
+    // never the configured reason (PR #1053 contract) — and the owner stays unchanged.
+    String clientTitle = coAccess ? clientCopy.coAccessGrantedTitle() : clientCopy.grantedTitle();
+    String clientDescription =
+        coAccess
+            ? renderClientCopy(clientCopy.coAccessGrantedDescription(), requesterName)
+                .replace(
+                    "{{duration}}",
+                    formatDuration(
+                        request.getMaxAccessDurationMinutes(), resolveSessionLanguage(session)))
+            : renderClientCopy(clientCopy.grantedDescription(), requesterName);
```
[the remark this answers is not in the document]
Comment (Storypapst) [reply src/main/java/de/caritas/cob/userservice/api/service/CaseHandoverService.java]: Grant notifications now render the TenantService-resolved `clientNotificationTemplates` (session language, `de` fallback, `{{newAdvisor}}`/`{{duration}}` tokens), with the built-in copy only as fallback - covered by a new test. The consent request deliberately keeps the generic copy: the per-reason template is grant wording by design (the defaults read "Du hast ... zugestimmt"), and the PR #1053 privacy contract keeps reason-derived text out of the pre-consent prompt.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Comment by Storypapst** [comment]
## Review round 2026-08-31

All nine unresolved review threads are addressed and resolved (details on each thread):

- **Policy cache**: refresh lease is environment-driven (`case.handover.policy-refresh-claim-duration`, default PT1M); a lease conflict without a snapshot now returns a retryable **503** instead of a 500; `getEffective` and the scheduled sweep open the transaction at the proxy boundary, with entry-point and annotation tests proving it.
- **Matrix safety**: a rolled-back grant now compensates the Matrix join it performed (transaction synchronization; pre-existing ADR-002 department memberships are never touched) - three new tests.
- **Changeset 0088**: the `OTHER_EMERGENCY` retirement only touches rows still carrying the untouched 0057/0059 seed values and is exactly reversible in rollback.
- **Tenant client templates**: grant notifications render the TenantService-resolved `clientNotificationTemplates` with the built-in client-safe copy as fallback; the pre-consent prompt stays generic per the PR #1053 privacy contract.
- **Tests**: expiry sweep proves continued processing past a Matrix failure; opt-out decline proves the actual `removeUserFromRoom` call.

**On the outside-diff finding "revoke Matrix membership at the expiry boundary":** the product-facing curtain closes exactly at the persisted `expiresAt` - `getStatus`/`toStatus` report `EXPIRED` and `canViewContent=false` from the first read past the deadline (covered by `getStatus_closesCoAccessExactlyAtExpiryEvenBeforeSweepRuns`). Matrix-level membership is reconciled by the leased sweep within the configured delay (60s default), which the replica-safety inventory documents as a bounded window; per-request timers would reintroduce the replica-coordination problems the lease design avoids. If the team wants a tighter bound, `case.handover.co-access-sweep-delay-ms` can be lowered without a code change.

**Merge freshness**: merged current `origin/pre-dev` into the branch (conflicts in `userservice-master.xml` include order and `replica-safety-components.json` resolved as unions; changeset numbers 0086-0088 remain free on pre-dev).

**CI**: the failing `openapi contract gate tests` job asserted the previous TenantService pin; the gate test now matches the workflow's `7b22058d` pin (green locally, 9 passed).

**Verification**: full unit suite on JDK 21 - **4094 tests, 0 failures**; `CaseHandoverServiceTest` 68 and `CaseHandoverPolicyCacheServiceTest` 9 all green; the new rollback-compensation test was mutation-checked (fails when the compensation is removed).

@coderabbitai review

_🤖 Generated with [Claude Code](https://claude.com/claude-code)_

**Comment by Storypapst** [comment]
Heads-up for the merge: pre-dev's DATABASECHANGELOG carried orphaned Aug-16 speed-deploy executions of a `0086-case-handover-co-access` / `0087-tenant-case-handover-policy-cache` changeset pair (from an ad-hoc image, never merged). The stale `0086`-prefixed rows were removed today while repairing a checksum clash with the now-merged DPA-forward 0086 (backup in `userservice.zz_changelog_backup_20260901` on pre-dev). Consequences for this PR: its changesets must be renumbered past the now-taken 0086/0087 at merge, and their DDL should tolerate tables possibly left behind by that Aug-16 execution (IF NOT EXISTS or preconditions).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Comment by Storypapst** [comment]
## ⛔ Blocker before merge: this PR would show the advice seeker the wrong card

Found during a case-handover archaeology pass on 2026-09-01 (full analysis: ORISO-Admin#771).

**The defect.** This PR adds a 6-arg `buildCaseHandoverParams` overload carrying `clientConsent`, plus a unit test asserting it serialises — but **no production call site passes it**. Verified by hunk audit: no hunk in this diff touches `notifyPendingConsent` (old lines 872–975).

Meanwhile the PR makes `notifyPendingConsent` reachable for OPT_OUT reasons (new line ~968). Downstream, `ORISO-Frontend/src/components/session/SessionStream.tsx:1181-1186` selects the card variant from `params?.clientConsent === 'OPT_OUT'`. With the field absent it falls through to the default branch — so an **opt-out** handover renders the **opt-in approve/decline card**.

That is the opposite of the intended behaviour and worse than today, where nothing is shown at all.

**Fix:** pass `clientConsent` from the resolved policy at the `notifyPendingConsent` call site. One line — but without it this PR is a user-visible regression.

## Two more things to settle before this merges

**1. It carries an unratified product change.** Beyond the consent plumbing this PR re-labels the handover reasons, disables `OTHER_EMERGENCY` (`accessAllowed(false).enabled(false)`), converts *Advice-Needed* from a takeover into 3-hour read-only co-access, and rewrites all client copy around `{{duration}}`. Nobody asked for that in the consent work. Either accept it deliberately or split consent-mode persistence into a focused ~200-line PR — which is probably the faster path to the Admin's five options actually persisting.

⚠️ **`{{duration}}` has no substitution site anywhere in the org.** The only substitution is `CaseHandoverService.renderClientCopy` for `{{newAdvisor}}`. If that copy ships, advice seekers see the literal token.

**2. It pre-decides an open privacy question.** `resolveTenantClientTemplate` routes tenant-authored templates to advice seekers, reopening the surface that PR #1053 (issue #1051) deliberately closed to stop the handover *reason* leaking. That ruling belongs to the owner and must land **before** this merges, not after — see ORISO-Admin#771 §9.3.

## Ordering

This PR **hard-depends on ORISO-TenantService#193** at runtime: it vendors `services/tenantadminservice.yaml`, so it compiles alone but is undeployable without #193's API. And do **not** land ORISO-TenantService#231 first — it touches the same two production files #193 rewrites and would add conflicts to an already-conflicting PR.

Note also: every review on this PR and its three siblings is from `coderabbitai`. There is no human sign-off in this chain yet.

Moved to **In progress** on the ORISO Roadmap; tracking issues #200/#201 set to **v2.0.4**.

**Comment by shazia-k** [comment]
Closing as superseded, together with TenantService #193 — the two pinned each other by SHA in the OpenAPI contract gate, so they were always going to move as a pair.

## Why it cannot be merged

Unlike #193, `dev` has **none** of the co-access work — `services/tenantadminservice.yaml` on `dev` has no permission-policy section, and no `dev` code calls `getTenantPermissionPolicies`. So almost all of this is genuine delta and worth keeping.

The problem is the other direction: the branch is 32 commits behind and its `CaseHandoverService.java` diff (+663/−62) **reverts work that landed on `dev`** in the meantime.

| Would be reverted | Consequence |
|---|---|
| Department (agency × topic) eligibility scoping, `feature.topics.enabled` (#202) | Falls back to plain same-agency — **widens who may request a handover**, a least-privilege regression no test on this branch would catch |
| `attachStandingSupervisorAfterCommit` (ADR-008) | Absent from this branch entirely — a replay **silently removes auto-supervision from every handover** |
| `SmtpSendException` category detail (#1006) | Error detail lost |
| Two `team-discussion` scheduler entries in `replica-safety-components.json` | Deleted from the inventory |

This is why it has to be re-derived on top of `dev` rather than merged or rebased. A rebase would replay the same semantic conflicts once per commit across 14 commits.

## Liquibase collision

`0086_case_handover_co_access`, `0087_tenant_case_handover_policy_cache` and `0088_case_handover_consent_mode` all collide with `dev`'s live `0086_dpa_forward_and_signed_notice`, `0087_account_invite_dpa_forward_count` and `0088_support_room_repair`. `dev` is occupied through `0090`.

Renumber to **`0091` / `0092` / `0093`**. The `0088` SQL's `WHERE code = 'OTHER_EMERGENCY' AND policy_authority = 'platform-admin-default-case-handover-policy'` still matches `dev`'s `0059` seed, so it carries over unchanged.

## One decision the re-cut must make

The branch gates consultant reassignment on `AccessType.TAKEOVER`; `dev` reassigns unconditionally and calls `attachStandingSupervisorAfterCommit`.

**Recommendation: move the supervisor attach inside the same `TAKEOVER` branch.** A CO_ACCESS grant never changes `session.consultant`, so attaching the requester's standing supervisor would add oversight to a case whose owner never changed — and nothing detaches it on expiry, because `dev`'s design never detaches.

## Re-cut scope → UserService #200 and #201

Three PRs, applied **on top of** `dev`'s department scoping and supervisor work:

1. **#201** — tri-state consent mode: `CaseHandoverConsentMode`, `clientConsent` on `CaseHandoverReasonPolicy` and `CaseHandoverRequest`, changeset `0093`. *(S–M)*
2. **#200** — CO_ACCESS: `AccessType`, `GRANTED_PENDING_CLIENT_OPTOUT`/`EXPIRED`, `expires_at`, the two pessimistic-lock finders, leased expiry sweep, Matrix revocation on declined OPT_OUT, changeset `0091`. *(M)*
3. **#201** — tenant policy cache client: `TenantCaseHandoverPolicyCache` + `CaseHandoverPolicyCacheService` (last-known-good, `stale_since`, never a permissive fallback), `ServiceUnavailableException` + 503 handler, `tenantadminservice.yaml` policy endpoint, changeset `0092`. *(M)*

Excluded: the `openapi-contracts.yml` cross-repo SHA pins and their gate assertions — dropping that coupling is the point of the re-cut. Also drop the hunk deleting `dpaVersion` from the DPA schema; `dev` added it.

Combined estimate **L (6–10 person-days)**.

## Not a breaking change

Neither UserService nor Frontend reads TenantService's policy shape, so the tri-state enum breaks only `ORISO-Admin/src/types/permissionPolicy.ts`, and only at the TypeScript level — the deprecated boolean mirror keeps the wire payload compatible.

Worth knowing: **Admin on `dev` already ships an OPT_OUT-capable UI while UserService can only store a boolean, so an OPT_OUT selection silently degrades to NONE today.** This work closes a gap that is already live rather than opening one.

The branch is not deleted; every commit remains at `56ee20d0`.
